cloudsim-plus/src/main/java/org/cloudsimplus/faultinjection/HostFaultInjection.java

```diff
@@ -450,101 +451,135 @@ private void setVmToFailedAndCreateClone(Vm vm) {
         if (Host.NULL.equals(lastFailedHost)) {
             return;
         }
-        
-        final double recoveryTime = randomRecoveryTimeForVm.sample();
-        Log.printFormattedLine("\t# Time to recovery the fault: %.2f minutes", recoveryTime/60);    
 
-        vmRecoveryTimeMap.put(vm, recoveryTime);
```

Comment:
This `vmRecoveryTimeMap.put(vm, recoveryTime);` must be moved for the first line inside the if below, because just there a clone is created to recovery from the failure.